ledger/simulation/simulation_eval_test.go

```diff
@@ -2297,6 +2490,627 @@ byte "hello"; log; int 0`,
 	})
 }
 
+const FrameBuryDigProgram = `#pragma version 8
+txn ApplicationID      // on creation, always approve
+bz end
+
+txn NumAppArgs
+int 1
+==
+assert
+
+txn ApplicationArgs 0
+btoi
+callsub subroutine_manipulating_stack
+itob
+log
+b end
+
+subroutine_manipulating_stack:
+  proto 1 1
+  int 0               // [0]
+  dup                 // [0, 0]
+  dupn 4              // [0, 0, 0, 0, 0, 0]
+  frame_dig -1        // [0, 0, 0, 0, 0, 0, arg_0]
+  frame_bury 0        // [arg_0, 0, 0, 0, 0, 0]
+  dig 5               // [arg_0, 0, 0, 0, 0, 0, arg_0]
+  cover 5             // [arg_0, arg_0, 0, 0, 0, 0, 0]
+  frame_dig 0         // [arg_0, arg_0, 0, 0, 0, 0, 0, arg_0]
+  frame_dig 1         // [arg_0, arg_0, 0, 0, 0, 0, 0, arg_0, arg_0]
+  +                   // [arg_0, arg_0, 0, 0, 0, 0, 0, arg_0 * 2]
+  bury 7              // [arg_0 * 2, arg_0, 0, 0, 0, 0, 0]
+  popn 5              // [arg_0 * 2, arg_0]
+  uncover 1           // [arg_0, arg_0 * 2]
+  +                   // [arg_0 * 3]
+  retsub
+
+end:
+  int 1
```

Comment:
Two suggestions for this test:
* Can you add `return` at the end?
* Can you add `swap` somewhere?

Reply:
I appended a `return` at the end, and added a `swap` after `uncover 1` in [per pr response](https://github.com/algorand/go-algorand/pull/5440/commits/e02b34c91c659836b3aec983577be3bb898cf09a)